apiserver/facades/agent/provisioner/provisioner.go

```diff
@@ -839,24 +839,86 @@ func (api *ProvisionerAPI) ReleaseContainerAddresses(ctx context.Context, args p
 
 // PrepareContainerInterfaceInfo allocates an address and returns information to
 // configure networking for a container. It accepts container tags as arguments.
-func (api *ProvisionerAPI) PrepareContainerInterfaceInfo(ctx context.Context, args params.Entities) (
-	params.MachineNetworkConfigResults,
-	error,
-) {
-	return api.prepareOrGetContainerInterfaceInfo(ctx, args, false)
-}
+func (api *ProvisionerAPI) PrepareContainerInterfaceInfo(
+	ctx context.Context, args params.Entities,
+) (params.MachineNetworkConfigResults, error) {
+	result := params.MachineNetworkConfigResults{}
+
+	canAccess, err := api.getAuthFunc(ctx)
+	if err != nil {
+		return result, errors.Trace(err)
+	}
+
+	hostAuthTag := api.authorizer.GetAuthTag()
+	if hostAuthTag == nil {
+		return result, apiservererrors.ErrPerm
+	}
+
+	if !canAccess(hostAuthTag) {
+		return result, apiservererrors.ErrPerm
+	}
+
+	hostName := coremachine.Name(hostAuthTag.Id())
+	hostUUID, err := api.machineService.GetMachineUUID(ctx, hostName)
+	if errors.Is(err, machineerrors.MachineNotFound) {
+		return result, apiservererrors.ServerError(errors.NotFoundf("machine %q", hostName))
+	}
+
+	hostInstanceID, err := api.machineService.GetInstanceID(ctx, hostUUID)
+	if errors.Is(err, machineerrors.NotProvisioned) {
+		return result, apiservererrors.ServerError(errors.NotProvisionedf("host machine %q", hostName.String()))
+	}
+
+	results := make([]params.MachineNetworkConfigResult, len(args.Entities))
+	for i, entity := range args.Entities {
+		gTag, err := names.ParseMachineTag(entity.Tag)
```

Comment:
suggestion: split the interior of the for loop into a new method, the current is quite long.

Reply:
Shortened it a different way. It's only 60 lines and the inner loop includes a large comment.